### ISSUE TITLE
Why not open source so others improve the code?

### ISSUE THREAD

**Issue by jimbo1qaz** [droidmjt/Droidsound#269]
It seems like you're withholding the updated source code from the repo. Is there any reason you don't want to release it?

If you do, it could help other people debug and fix accumulated issues, as well as implement long-running and new feature requests you don't have time for (or don't care about). Particularly, the pausing state machine bugs, multiple-folder playback, reducing memory usage for play queues, etc.

(When you preload all queued files, does that mean you really load hundreds of files, each containing megabytes, into memory?)

I'd like to help, though I can't do much with two-year-old unlicensed code, and I really don't want to resort to writing my own player.

In short, open-sourcing the program can only accelerate development and improve the program.

**Comment by crimson1117** [comment]
I remembered the basic code of Droidsound is by swimsuitboys.

**Comment by droidmjt** [comment]
>>(When you preload all queued files, does that mean you really load hundreds of files, each containing megabytes, into memory?)

I don't do such thing. I preload nothing. It only queues the filenames, thats all.

**Comment by Level99OCR** [comment]
While I no longer have any personal delusions about getting my code up to snuff to help out with this project, I can understand jimbo's reasoning. Droidsound E, which originally started as a branch of the opensource project Droidsound from SwimsuitBoys, including numerous opensource plugin packs and the like, having been heavily worked on for the past three or four years here with a two year outdated source code seems a little wonky.

Not sure what your concerns are with updating the source code, MJT, but I'm sure they're reasonable. It would be worth it to take into consideration, now as opposed to when this project started and almost nobody was able to contribute code and there were far few bug testers, to consider getting back into updating the source here. You could keep working on your own personal to-do list of features you want to see implemented, while others motivated to fix issues they find if possible can do just that.

If you're worried about doing a code cleanup from having things be super-sloppy and poorly documented, I would say that having some comments with updated code are better than none, even if its not what it should be.

You've worked hard to make this what it is, and I'm sure you'll continue doing so for a good while into the future, but do consider updating the source at some point so that others can jump in and help now. You've carried this burden on your back by yourself for a while, I think you deserve any helping hands you can get :D 

I know you did say to someone else that, if they were interested in tackling some part of the code, to contact you on the side. Though I guess that kind of defeats the spirit of having things here on github in the first place?

I'm just tossing out ideas, dude. You do what you gotta do, I'm happy just being a tester, but I can see others finding it a bit frustrating to have even the deprecated code here. Github has been great for reporting bugs and getting things fixed, regardless. If it can be taken fuller advantage of by updating the source files, more reason to do so!

**Comment by droidmjt** [comment]
I have personal reason for not sharing the source.

**Comment by jimbo1qaz** [comment]
"personal reason"

That seems somewhat shady...? Stolen code, license violations, feel superior to others?

**Comment by droidmjt** [comment]
for crying out loud, you can see what plugins I use, they are all public, so no, no license violations or stolen code

**Comment by jimbo1qaz** [comment]
well, what do you lose from open-source?

**Comment by droidmjt** [comment]
why do you need it so badly?

**Comment by crimson1117** [comment]
......(seems vgmstream still not support hca yet TAT)

**Comment by jimbo1qaz** [comment]
1. The freedom to see what's in there. And fix annoyances without watching them pile up in the issue tracker for weeks.
2. Add new features. Example: looped audio files (outside of vgmstream). I put loop points in the file name, I wrote an online player to play looped content. It's a bit tricky, and I don't expect anyone but me would care about it. But it's a nice feature, especially for a game-focused player like DSE.
    * https://jimbo1qaz.github.io/loopplayer/?file=sounds/ff6decisive.yaml
    * https://github.com/jimbo1qaz/loopplayer/
3. (When was the last time you made an official release? Almost nobody can find the improvements here.)

In other words, open-sourcing the project will produce a better player, with more people able to contribute, and with less bugs piled up, less work for you to fix.

Not open sourcing... Well, you'll be the only person to see the code. Do you need the code for your job?

**Comment by crimson1117** [comment]
Almost nobody can find the improvements here.
>http://vgmrips.net/packs/
Android	Droidsound>https://github.com/droidmjt/Droidsound/
ummmmmmmm,at least the guys who made vgm packs(including me)know this place well.
also guys from hcs64.

**Comment by xperia64** [comment]
I don't want to be "that guy", but unless the source is released, this project is violating the GPL.
Binaries which are publicly available without the corresponding source code are not permitted.
GPL components in DroidSound include:
Sidplayfp
vio2sf
lazyusf
id3tag
HTPlugin
playgsf
Specific UADE modules

Including GPL components in a project directly means that the project itself must be GPL.

**Comment by kode54** [comment]
I am being asked to weigh in my opinion on this matter, since so many of the libraries included are maintained by me. I don't remember whether I gave permission, but I have received code back, in the form of some Neon optimizations for lazyusf.

While I would like to see updated source code, it doesn't really bother me that much, as I don't really care to do my own development on Android, short of the foobar2000 Mobile project, which itself will probably be some sort of GPL violation powerhouse because it will be strictly commercial with no source code published, so long as it contains any of my GPLed libraries. Well, assuming the original maintainers don't care if I unilaterally give permission to use the code under a proprietary license.

**Comment by kode54** [comment]
On second thought, I would like to see the source to this again some day. I don't even understand the reason for keeping it private, unless it is planned to release the Play Store version as a paid app. Otherwise, why stifle competition from the upstream libraries?

**Comment by droidmjt** [comment]
i havent updated any libraries myself, the only library i have ever updated is the lazyusf2 neon thingy, thats all

**Comment by droidmjt** [comment]
i can upload the lazyusf2 code if you want to

**Comment by droidmjt** [comment]
actually, my bad, yes i have updated couple of libraries, i will upload the changes now
